### ISSUE TITLE
`inlineDynamicImports: true` and `moduleSideEffects: 'no-treeshake'` causes a hard-to-diagnose error

### ISSUE THREAD

**Issue by lemonmade** [rollup/rollup#4098]
- Rollup Version: 2.49.0
- Operating System (or Browser): macOS 11.3.1
- Node Version (if applicable):
- Link to reproduction (IMPORTANT, read below): https://replit.com/@lemonmade/rollup-tree-shaking-error-reproduction (repo version: https://github.com/lemonmade/rollup-treeshaking-issue)

### Expected Behavior

I can set `inlineDynamicImports: true`, and resolve a module with `moduleSideEffects: 'no-treeshake'`, and have the build complete without an error (in the example, removing `moduleSideEffects` or `inlineDynamicImports` both work, and neither changes the output, but my real project requires both to be enabled).

### Actual Behavior

The following error is thrown:

```
[!] TypeError: Cannot read property 'suggestedVariableName' of undefined
TypeError: Cannot read property 'suggestedVariableName' of undefined
    at Object.deconflictImportsOther (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:11071:80)
    at deconflictChunk (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:11038:52)
    at Chunk.setIdentifierRenderResolutions (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:12323:9)
    at Chunk.preRender (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:11707:14)
    at Bundle.prerenderChunks (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:13027:19)
    at Bundle.generate (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:12892:18)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async handleGenerateWrite (/home/runner/rollup-repro/node_modules/rollup/dist/shared/rollup.js:20623:23)
    at async Promise.all (index 0)
    at async build (/home/runner/rollup-repro/node_modules/rollup/dist/bin/rollup:1533:5)
```

I tracked this down a little bit, and noticed that we add the offending `undefined`s in https://github.com/rollup/rollup/blob/d73b2ace7aa352980920745690a01c0190d6f4a7/src/utils/getStaticDependencies.ts#L42-L46. Interestingly, the module that gets added here (e.g., has `chunkByModule.get(dependency) == null`) was always an index file that just re-exports other parts of the module being imported (I used a library from my real example, `@remote-ui/core`, but it was not specific to this library in my larger example).

Thanks in advance for your help, and for the amazing work on Rollup!

**Comment by lukastaegert** [comment]
Thanks for spotting, fix at #4104

**Comment by lemonmade** [comment]
@lukastaegert Thank you so much for the quick fix!
